File: src/main/java/org/aopbuddy/plugin/infra/config/ServerConfig.java
package org.aopbuddy.plugin.infra.config;

import com.intellij.openapi.components.PersistentStateComponent;
import com.intellij.openapi.components.ServiceManager;
import com.intellij.openapi.components.State;
import com.intellij.openapi.components.Storage;
import com.intellij.util.xmlb.XmlSerializerUtil;
import lombok.Data;
import org.aopbuddy.plugin.infra.model.HttpServer;
import org.jetbrains.annotations.NotNull;

import java.util.HashMap;
import java.util.Map;

@State(
        name = "ServerConfig",
        storages = @Storage("aopBuddyServerConfig.xml")
)
@Data
public class ServerConfig implements PersistentStateComponent<ServerConfig> {

    private Map<String, HttpServer> serverMap = new HashMap<>();

    public static ServerConfig getInstance() {
        return ServiceManager.getService(ServerConfig.class);
    }

    @Override
    public ServerConfig getState() {
        return this;
    }

    @Override
    public void loadState(@NotNull ServerConfig state) {
        XmlSerializerUtil.copyBean(state, this);
    }
}

File: src/main/kotlin/org/aopbuddy/plugin/infra/config/ServerConfig.kt
package org.aopbuddy.plugin.infra.config

import com.intellij.openapi.components.*
import org.aopbuddy.plugin.infra.model.HttpServer

@State(
    name = "ServerConfig",
    storages = [Storage("aopBuddyServerConfig.xml")]
)
@Service
class ServerConfig : PersistentStateComponent<ServerConfig> {

    var serverMap: MutableMap<String, HttpServer> = HashMap()

    override fun getState(): ServerConfig {
        return this
    }

    override fun loadState(state: ServerConfig) {
        this.serverMap.clear()
        this.serverMap.putAll(state.serverMap)
    }

    companion object {
        fun getInstance(): ServerConfig {
            return ServiceManager.getService(ServerConfig::class.java)
        }
    }
}